pkg/crc/services/dns/dns_darwin.go

```diff
@@ -123,37 +124,39 @@ func waitForNetwork(hostResolv *network.ResolvFileValues) error {
 	return fmt.Errorf("Host is not connected to internet.")
 }
 
-// updateResolvConfFile updates the host's /etc/resolv.conf file with Instance IP.
-func updateResolvConfFile(instanceIP string, resolvConfFile string) (bool, error) {
-	// Get the current value of resolv.conf file
-	hostResolv, err := network.GetResolvValuesFromHost()
+// updateHostsConfFile updates the host's /etc/hosts file with Instance IP.
+func updateHostsConfFile(instanceIP string, hostnames ...string) (bool, error) {
+	// Get the current value of /etc/hosts file
+	hosts, err := goodhosts.NewHosts()
 	if err != nil {
-		return false, fmt.Errorf("Unable to read host resolv file (%v)", err)
+		return false, err
 	}
-
-	foundExistingInstanceIP := false
-
-	for i, ns := range hostResolv.NameServers {
-		// Update the nameserver IP with instance IP if already exist
-		if strings.Contains(ns.IPAddress, "192.168.130") {
-			hostResolv.NameServers[i].IPAddress = instanceIP
-			foundExistingInstanceIP = true
-			break
+	for _, hostname := range hostnames {
+		if err := replaceHostname(hosts, instanceIP, hostname); err != nil {
+			return false, err
 		}
 	}
 
-	if !foundExistingInstanceIP {
-		nameserver := network.NameServer{IPAddress: instanceIP}
-		nameservers := []network.NameServer{nameserver}
-		hostResolv.NameServers = append(nameservers, hostResolv.NameServers...)
+	// Flush operation is required just after replacing the host this
+	// add the entry to /etc/hosts file
+	if err := hosts.Flush(); err !=nil {
+		return false, err
 	}
 
-	// Write to the resolv.conf file.
-	resolvFile, _ := network.CreateResolvFile(*hostResolv)
-	err = ioutil.WriteFile(resolvConfFile, []byte(resolvFile), 0644)
-	if err != nil {
-		return false, err
+	return true, nil
+}
+
+func replaceHostname(hosts goodhosts.Hosts, ipAddress string, hostname string) error {
```

Comment:
My bad, I hadn't tested these proposed changes besides compilation. This has to be `hosts *goodhosts.Hosts` otherwise /etc/hosts will not be modified

Reply:
@cfergeau it is modified, what actually happened in your proposed changes you didn't try to use `host.Flush()` which actually write the file to disk, I already do that so this is not required.